From e760564eab4dd0ef27f01547b9d737cfd4dd849c Mon Sep 17 00:00:00 2001
From: Heinrich Toews <heinrich.toews@wago.com>
Date: Wed, 13 Mar 2019 16:34:39 +0100
Subject: [PATCH] spi: omap2: allow turbo mode to be used also in TX mode

This fix was necessary in order to make TURBO mode also work
in TX transfers. The TURBO mode function was tested very well in
combination of the anv32aa1w driver and it showed to
work very stable. The amount of data transfered was 128k.

Through tubro mode the inter-word GAP was reduced from 320ns to only 52ns.

Signed-off-by: Heinrich Toews <heinrich.toews@wago.com>
---
 drivers/spi/spi-omap2-mcspi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/spi-omap2-mcspi.c b/drivers/spi/spi-omap2-mcspi.c
index e9192b6..fdadb38 100644
--- a/drivers/spi/spi-omap2-mcspi.c
+++ b/drivers/spi/spi-omap2-mcspi.c
@@ -1325,7 +1325,7 @@ static int omap2_mcspi_transfer_one(struct spi_master *master,
 	else if (t->rx_buf == NULL)
 		chconf |= OMAP2_MCSPI_CHCONF_TRM_TX_ONLY;
 
-	if (cd && cd->turbo_mode && t->tx_buf == NULL) {
+	if (cd && cd->turbo_mode) {
 		/* Turbo mode is for more than one word */
 		if (t->len > ((cs->word_len + 7) >> 3))
 			chconf |= OMAP2_MCSPI_CHCONF_TURBO;
-- 
2.7.4

